src/main.rs

```diff
@@ -1,3 +1,186 @@
-fn main() {
-    println!("Hello, world!");
+use actix_web::{web, App, HttpServer, Result};
+use serde::Deserialize;
+extern crate serde_json;
+use serde_json::Value;
+
+//extern crate reqwest;
+use reqwest;
+// This struct represents state
+
+#[derive(Deserialize)]
+struct ContractCall {
+    contract: String,
+    txn: String,
+}
+
+async fn get_txn_info(txn: &String) -> String {
+    let res = reqwest::get(format!(
+        "https://blockscout.com/eth/mainnet/api?module=transaction&action=gettxinfo&txhash={}",
+        txn
+    ))
+    .await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    let res = res.text().await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    res
+}
+
+fn get_txn_input_bytes(txn: &String) -> String {
+    let data: Value = serde_json::from_str(&txn).unwrap();
+    let obj = data.as_object().unwrap();
+    let s = String::from(format!(
+        "{}",
+        obj.get("result").unwrap().get("input").unwrap()
+    ));
+    let res = &s[1..11];
+    res.to_string()
+}
+
+async fn get_txn_method(input: &String) -> String {
+    //https://www.4byte.directory/signatures/?bytes4_signature=0xa7ca0ba3
+    let res = reqwest::get(format!(
+        "https://www.4byte.directory/api/v1/signatures/?hex_signature={}",
+        input
+    ))
+    .await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    let res = res.text().await;
+
+    let res = match res {
+        Ok(r) => r,
+        Err(err) => panic!("{}", err),
+    };
+
+    res
+}
+
+fn decode_methods(methods: &String) -> Vec<String> {
+    let data: Value = serde_json::from_str(&methods).unwrap();
+    let obj = data.as_object().unwrap();
+    let cnt = obj.get("count").unwrap();
+    println!("{}", cnt);
```

Comment:
I understand that this is work in progress, but please, no printlns in production code :) 
if you want to log something, use `log` crate and proper level (info/debug/etc)